ts/main.ts

```diff
@@ -1,61 +1,179 @@
+  const title = document.createElement('h2');
+  title.textContent = entry.title;
+  const notes = document.createElement('p');
+  notes.textContent = entry.notes;
+
+  textColumn.appendChild(title);
+  textColumn.appendChild(notes);
+
+  // append both image and text columns
+  li.appendChild(imgColumn);
+  li.appendChild(textColumn);
+
+  return li;
+}
+
+// toggleNoEntries function
+function toggleNoEntries(): void {
+  const noEntriesMessage = document.querySelector(
+    '#no-entries-message',
+  ) as HTMLElement;
+
+  if (data.entries.length === 0) {
+    noEntriesMessage.classList.remove('hidden');
+  } else {
+    noEntriesMessage.classList.add('hidden');
+  }
+}
+
+// handle form submission
+function handleFormSubmit(event: Event): void {
   event.preventDefault();
 
-  // new entry object from the form and assign entryID
+  // create a new entry
   const newEntry: Entry = {
     entryID: data.nextEntryId,
     title: $titleInput.value,
     imgUrl: $imgUrlInput.value,
     notes: $notesInput.value,
   };
 
-  // increment the nextEntryID
+  // increment entry ID
   data.nextEntryId++;
 
-  // add new entry to beginning of array of entries
+  // add the new entry to the array
   data.entries.unshift(newEntry);
 
-  // save updated data to localStorage
+  // render the new entry in the list
+  const entriesList = document.querySelector(
+    '[data-view="entries"] ul',
+  ) as HTMLUListElement;
+  const newEntryLi = renderEntry(newEntry);
+  entriesList.prepend(newEntryLi);
+
+  // display the entries view
+  viewSwap('entries');
+
+  // update the "no entries" message visibility
+  toggleNoEntries();
+
+  // save the updated data
   writeData();
 
-  // clear and reset form fields
-  $imgPreview.src = './images/placeholder-image-square.jpg';
-  $imgPreview.alt = 'default placeholder image for journal entry';
+  // clear and reset the form
   $form.reset();
+
+  // reset the image preview
+  $imgPreview.src = './images/placeholder-image-square.jpg';
+  $imgPreview.alt = 'Preview image';
+}
+
+// handle image preview input
+function handleImagePreview(): void {
+  const url = $imgUrlInput.value;
+  // only update the image if the URL is valid
+  if (url) {
+    $imgPreview.src = url;
+    $imgPreview.alt = 'Preview image';
+  }
+}
+
+// event listeners
+document.addEventListener('DOMContentLoaded', () => {
+  // Show the previously active view
+  const previousView = data.view || 'entries';
+  viewSwap(previousView);
+
+  // toggle "no entries" message
+  toggleNoEntries();
+
+  // render existing entries on load
+  const entriesList = document.querySelector(
+    '[data-view="entries"] ul',
+  ) as HTMLUListElement;
+  data.entries.forEach((entry) => {
+    const entryLi = renderEntry(entry);
+    entriesList.appendChild(entryLi);
+  });
 });
+
+// Event listener for the "New Entry" button
+const newEntryButton = document.getElementById('new-entry-btn')!;
+newEntryButton.addEventListener('click', (event) => {
+  event.preventDefault();
```

Comment:
This is really only needed on `submit` events to prevent the page from reloading. Other than that, I never include this in an event